src/water_cmd.cpp

```diff
@@ -926,12 +926,12 @@ static void DrawTile_Water(TileInfo *ti)
 	switch (GetWaterTileType(ti->tile)) {
 		case WATER_TILE_CLEAR:
 			DrawWaterClassGround(ti);
-			DrawBridgeMiddle(ti);
+			DrawBridgeMiddle(ti, {BridgePillarFlag::EdgeSW, BridgePillarFlag::EdgeNE, BridgePillarFlag::EdgeNW, BridgePillarFlag::EdgeSE});
```

Comment:
Bridges over ocean only get pillars if they're in the tile corners? That seems harsh. 🙂 

I'd rather see diagonally-moving ships clip through bridge pillars than see no pillars for some (NewGRF) bridge types.